tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/replication/ReplicationTest.java

```diff
@@ -363,6 +363,60 @@ public void testSendPacketsWithFailure() throws Exception {
       }
    }
 
+   @TestTemplate
+   public void testSendMessageWithLargeHeader() throws Exception {
+      setupServer(true, TestInterceptor.class.getName());
+
+      manager = liveServer.getReplicationManager();
+      waitForComponent(manager);
+
+      ClientSessionFactory sf = createSessionFactory(locator);
+
+      try (ClientSession producerSession = sf.createSession();
+           ClientSession consumerSession = sf.createSession()) {
+
+         producerSession.start();
+         producerSession.createQueue(QueueConfiguration.of(ADDRESS));
+         ClientProducer producer = producerSession.createProducer(ADDRESS);
+
+         consumerSession.start();
+         ClientConsumer consumer = consumerSession.createConsumer(ADDRESS);
+
+         ClientMessage messageBefore = producerSession.createMessage(true);
+         setBody(0, messageBefore);
+         messageBefore.putIntProperty("counter", 0);
+         producer.send(messageBefore);
+
+         ClientMessage messageReceivedBefore = consumer.receive(1000);
+         assertNotNull(messageReceivedBefore, "Message should exist!");
+         assertMessageBody(0, messageReceivedBefore);
+         assertEquals(0, messageReceivedBefore.getIntProperty("counter").intValue());
+         messageReceivedBefore.acknowledge();
+
+         ClientMessage messageWithLargeHeader = producerSession.createMessage(true);
+         setBody(1, messageWithLargeHeader);
+         messageWithLargeHeader.putIntProperty("counter", 1);
+         messageWithLargeHeader.putStringProperty("large-property", "z".repeat(512 * 1024));
+         try {
+            producer.send(messageWithLargeHeader);
+            fail();
+         } catch (Exception e) {
+            assertTrue(e.getMessage().contains("AMQ149005"));
+         }
+
+         ClientMessage messageAfter = producerSession.createMessage(true);
+         setBody(2, messageAfter);
+         messageAfter.putIntProperty("counter", 2);
+         producer.send(messageAfter);
+
+         ClientMessage messageReceivedAfter = consumer.receive(1000);
+         assertNotNull(messageReceivedAfter, "Message should exist!");
+         assertMessageBody(2, messageReceivedAfter);
+         assertEquals(2, messageReceivedAfter.getIntProperty("counter").intValue());
+         messageReceivedAfter.acknowledge();
```

Comment:
small nitpick... 

assertNull(consumer.receiveNoWait());

to make sure the first one was not sent on the replica?

Reply:
It's such a small thing.. I'm doing it myself and amending it as your commit...

Reply:
thanks